__tests__/integration.test.js

```diff
@@ -0,0 +1,46 @@
+const db = require("../db/connection");
+const seed = require("../db/seeds/seed");
+const testData = require("../db/data/test-data");
+const request = require("supertest");
+const app = require("../app");
+
+beforeEach(() => seed(testData));
+afterAll(() => db.end());
+
+
+//Invalid Endpoint
+
+describe("/*", () => {
+
+  test("Returns a 400 error for invalid endpoints", () => {
+    return request(app)
+      .get("/api/path-doesnt-exist")
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("Invalid path");
+      });
+  });
+  
+});
+
+
+//api/topics
+
+describe("/api/topics", () => {
+
+  test("GET: Requests to this endpoint should respond with an array of topic objects, each with slug and description properties", () => {
+    return request(app)
+    .get("/api/topics")
+    .expect(200)
+    .then(({body})=>{
+      const topics = body
+      expect(topics.length).toBe(3)
+      topics.forEach(topic => {
+        expect(typeof topic.slug).toBe("string")
```

Comment:
this is fine, but I would recommend looking into the jest matchers .toMatchObject and .objectContaining and having a go at using them for future tickets (particularly when you'll be testing against more complicated objects)